File: game-server/src/main/java/org/forfun/mmorpg/game/cross/router/RoundBalanceStrategy.java
package org.forfun.mmorpg.game.cross.router;


import org.springframework.util.CollectionUtils;

import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

public class RoundBalanceStrategy implements BalanceStrategy {

    private AtomicInteger counter = new AtomicInteger();

    @Override
    public int next(List<Integer> nodes) {
        if (CollectionUtils.isEmpty(nodes)) {
            return 0;
        }
        return nodes.get(counter.getAndIncrement() % nodes.size());
    }

}

File: game-server/src/main/java/org/forfun/mmorpg/game/cross/router/RoundBalanceStrategy.kt
package org.forfun.mmorpg.game.cross.router

import org.springframework.util.CollectionUtils
import java.util.concurrent.atomic.AtomicInteger

class RoundBalanceStrategy : BalanceStrategy {

    private val counter = AtomicInteger()

    override fun next(nodes: List<Int>): Int {
        if (CollectionUtils.isEmpty(nodes)) {
            return 0
        }
        return nodes[counter.getAndIncrement() % nodes.size]
    }
}
